src/index.js

```diff
@@ -1 +1,124 @@
 'use strict';
+/* eslint-disable no-console */
+
+const http = require('http');
+const path = require('path');
+const fs = require('fs');
+const { IncomingForm } = require('formidable');
+
+const PORT = process.env.PORT || 8080;
+const server = new http.Server();
+
+server.on('request', (req, res) => {
+  if (req.url === '/data' && req.method.toLowerCase() === 'post') {
+    const form = new IncomingForm();
+
+    form.parse(req, (err, fields) => {
+      if (err) {
+        res.statusCode = 400;
+        res.end(err);
+
+        return;
+      }
+
+      const { date, title, amount } = fields;
+
+      if (!date || !title || !amount) {
+        res.statusCode = 400;
+        res.end('All fields are required.');
+
+        return;
+      }
+
+      const formData = {
+        date: date[0],
+        title: title[0],
+        amount: amount[0],
+      };
+
+      const filepath = path.join(__dirname, 'history', `${date[0]}.json`);
+
+      const formDataStream = fs.createWriteStream(filepath);
+
+      formDataStream.on('error', () => {
+        res.statusCode = 500;
+        res.end('Server error');
+      });
+
+      formDataStream.write(JSON.stringify(formData, null, 2));
+      formDataStream.end();
+
+      res.statusCode = 200;
+      res.setHeader('Content-Type', 'text/html');
+
+      res.end(`
+        <!DOCTYPE html>
```

Comment:
could you move it to some separate constant or function, accept formData, and return this string. To simplify code in your general function